### PR TITLE
Made fruit right click-able

### PR DESCRIPTION
**Pull request by Sfiguz7**
## Description
Made fruit drop on right click.

## Changes
Simply moved the block of code dropping fruit to a method in ExoticGarden, then called it in both Event listens. Had to change return value for the breaking leaves cycling.
Closed and reopened to copy paste over a better template (no need to I know but eh too late)

## Related Issues
N/A, approved suggestion on Discord.

## Checklist
- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 1]
```diff
@@ -892,6 +893,23 @@ public static ItemStack harvestPlant(Block block) {
...
+		Location loc = fruit.getLocation();
+		SlimefunItem check = BlockStorage.check(loc);
+		if (check == null) return;
+
+		for (Tree tree : ExoticGarden.getTrees()) {
+			if (check.getID().equalsIgnoreCase(tree.getFruitID())) {
+				BlockStorage.clearBlockInfo(loc);
+				ItemStack fruits = check.getItem();
+				fruit.getWorld().playEffect(loc, Effect.STEP_SOUND, Material.OAK_LEAVES);
+				fruit.getWorld().dropItemNaturally(loc, fruits);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:906]: Don't drop the original, drop a .clone()

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 2]
```diff
@@ -892,6 +893,23 @@ public static ItemStack harvestPlant(Block block) {
 		return null;
 	}
 
+	public static void harvestFruit(Block fruit) {
+		Location loc = fruit.getLocation();
+		SlimefunItem check = BlockStorage.check(loc);
+		if (check == null) return;
+
+		for (Tree tree : ExoticGarden.getTrees()) {
+			if (check.getID().equalsIgnoreCase(tree.getFruitID())) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:902]: This should REEEAAAALLLY be a HashMap lookup tbh.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java (hunk 1 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:906]: Boi I literally moved your code >:( It has always worked with this code and still does
Uff I'll change it

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java (hunk 2 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:902]: Same as above, your code not mine ahaha Define HashMap lookup, do you want me to save a HashMap with all fruit IDs somewhere on enable and check against that?

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java (hunk 1 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:906]: Something being my code is not a quality indicator XD

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java (hunk 2 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:902]: Yes

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java (hunk 1 above)
Comment (SoSeDiK) [reply src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java]: > Don't drop the original, drop a .clone()

CraftBukkit already creates a copy before spawning an item via `CraftItemStack.asNMSCopy(item)`, so it should be fine as long as if we don't do anything else to this item in our code. But yeah, it's true only for this specific case, and the item normally should be cloned normally.

**Comment by TheBusyBiscuit** [comment]
Oh right, this PR exists too XD

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 3]
```diff
@@ -892,6 +898,22 @@ public static ItemStack harvestPlant(Block block) {
...
 	}
 
+	public static void harvestFruit(Block fruit) {
+		Location loc = fruit.getLocation();
+		SlimefunItem check = BlockStorage.check(loc);
+		if (check == null) {
+		    return;
+        }
+
+        if (treeFruits.containsValue(check.getID())) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:908]: Oh i see, you are only using it for a .contains() check anyway, in that case a HashSet would probably be most optimal.

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 4]
```diff
@@ -892,6 +898,22 @@ public static ItemStack harvestPlant(Block block) {
 		return null;
 	}
 
+	public static void harvestFruit(Block fruit) {
+		Location loc = fruit.getLocation();
+		SlimefunItem check = BlockStorage.check(loc);
+		if (check == null) {
+		    return;
+        }
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:906]: See you in indentation hell

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by Sfiguz7** [comment]
Finally got to it, I think these two small fixes were all I was missing. Lmk

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 5]
```diff
@@ -208,10 +209,14 @@ public void onEnable() {
             String key = iterator.next();
             cfg.setDefaultValue("grass-drops." + key, true);
             if (!cfg.getBoolean("grass-drops." + key))
-                iterator.remove();
+                  iterator.remove();
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:212]: Imagine having 6 spaces PepeLaugh

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 6]
```diff
@@ -256,7 +262,7 @@ private void saveSchematic(@Nonnull String id) {
                 }
             }
         } catch (IOException e) {
-            getLogger().log(Level.SEVERE, e, () -> "Failed to load file: \"" + id + ".schematic\"");
+            ExoticGarden.getInstance().getLogger().log(Level.SEVERE, e, () -> "Failed to load file: \"" + id + ".schematic\"");
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:265]: You're getting the instance of this class??? why not just keep it as `getLogger`

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 7]
```diff
@@ -312,7 +320,9 @@ private void registerMagicalPlant(String name, ItemStack item, String texture, I
         Berry berry = new Berry(essence, upperCase + "_ESSENCE", PlantType.ORE_PLANT, texture);
         berries.add(berry);
 
-        new SlimefunItem(magicalCategory, new SlimefunItemStack(enumStyle + "_PLANT", Material.OAK_SAPLING, "&r" + name + " Plant"), RecipeType.ENHANCED_CRAFTING_TABLE, recipe).register(this);
+        new SlimefunItem(magicalCategory, new SlimefunItemStack(enumStyle + "_PLANT", Material.OAK_SAPLING, "&r" + name + " Plant"), RecipeType.ENHANCED_CRAFTING_TABLE,
+        recipe)
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:324]: errr

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 8]
```diff
@@ -329,38 +339,55 @@ public static ItemStack harvestPlant(@Nonnull Block block) {
         }
 
         for (Berry berry : getBerries()) {
-            if (item.getId().equalsIgnoreCase(berry.getID())) {
+            if (item.getID().equalsIgnoreCase(berry.getID())) {
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:342]: `getID()` is deprecated, use `getId()`
```suggestion
            if (item.getId().equals(berry.getId())) {
```

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java [hunk 9]
```diff
@@ -329,38 +339,55 @@ public static ItemStack harvestPlant(@Nonnull Block block) {
...
 
+    public void harvestFruit(Block fruit) {
+        Location loc = fruit.getLocation();
+        SlimefunItem check = BlockStorage.check(loc);
+
+        if (check == null) {
+            return;
+        }
+
+        if (treeFruits.contains(check.getID())) {
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:382]:
```suggestion
        if (treeFruits.contains(check.getId())) {
```

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java (hunk 8 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/exoticgarden/ExoticGarden.java:342]: nO I like it better and I'm a rebel deal with it

**Comment by TheBusyBiscuit** [comment]
merge conflicts

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**